packages/language-server/test/plugins/typescript/features/CompletionsProvider.test.ts

```diff
@@ -0,0 +1,125 @@
+
+		const completions = await provider.getCompletions(document, Position.create(0, 0));
+
+		expect(completions).to.be.null;
+	});
+
+	it('provide auto import completion with insert action for component - no front matter', async () => {
+		const { provider, document } = setup('autoImportNoFrontmatter.astro');
+
+		const completions = await provider.getCompletions(document, Position.create(1, 9));
+		const item = completions?.items.find((completion) => completion.label === 'Component');
+		const { additionalTextEdits, detail } = await provider.resolveCompletion(document, item!);
+
+		expect(detail).to.equal('./imports/component.astro');
+		expect(additionalTextEdits[0].newText).to.equal(
+			`---${newLine}import Component from './imports/component.astro'${newLine}---${newLine}${newLine}`
+		);
+	});
+
+	it('provide auto import completion with insert action for component - has front matter', async () => {
+		const { provider, document } = setup('autoImportFrontmatter.astro');
+
+		const completions = await provider.getCompletions(document, Position.create(4, 9));
+		const item = completions?.items.find((completion) => completion.label === 'Component');
+
+		const { additionalTextEdits, detail } = await provider.resolveCompletion(document, item!);
+
+		expect(detail).to.equal('./imports/component.astro');
+		expect(additionalTextEdits[0].newText).to.equal(
+			`${newLine}import Component from './imports/component.astro';${newLine}`
+		);
+	});
+
+	it('resolve completion without auto import if component import already exists', async () => {
+		const { provider, document } = setup('autoImportComponentAlreadyExists.astro');
+
+		const completions = await provider.getCompletions(document, Position.create(4, 9));
+
+		const item = completions?.items.find((completion) => completion.label === 'Component');
+
+		const { additionalTextEdits } = await provider.resolveCompletion(document, item!);
+
+		expect(additionalTextEdits).to.be.undefined;
+	});
+
+	it('provide import completions for supported files', async () => {
+		const { provider, document } = setup('importSupportedFormats.astro');
+
+		const completions = await provider.getCompletions(document, Position.create(1, 35));
+		const foundFiles = completions.items.map((completion) => completion.label);
+
+		expect(foundFiles).to.deep.equal(['Astro.astro', 'JSX', 'Svelte.svelte', 'Vue.vue', 'js', 'json.json', 'ts']);
+	});
+
+	it('provide completion inside import statement', async () => {
+		const { provider, document } = setup('importStatement.astro');
+
+		const completions = await provider.getCompletions(document, Position.create(1, 11), {
+			triggerKind: CompletionTriggerKind.Invoked,
+		});
+
+		const item = completions?.items.find((completion) => completion.label === 'MySuperFunction');
+		delete item.data;
+
+		expect(item).to.deep.equal({
+			commitCharacters: ['.', ',', '('],
+			insertText: "import { MySuperFunction$1 } from './imports/definitions';",
```

Comment:
The `$1` here is not an error, TypeScript does in fact position the user's cursor there after this kind of completion. 

I don't find this behavior necessarily good and fun to use, but it is consistent with what you would get inside .ts files